src/site/content/en/blog/javascript-and-google-search-io-2019/index.md

```diff
@@ -0,0 +1,169 @@
+
+Googlebot is a pipeline with different components. Let's take a look at this pipeline to understand how Googlebot indexes pages for Google Search.
+
+<figure class="w-figure w-figure--center w-figure--fullbleed">
+  <img src="googlebot-process.png" alt="A diagram showing a URL moving from a crawling queue to a processing step that extracts linked URLs and adds them to the crawling queue, a rendering queue that feeds into a renderer which produces HTML. The processor uses this HTML to extract linked URLs again and index the content.">
+  <figcaption class="w-figcaption w-figcaption--fullbleed">
+    Fig. 1 — Googlebot's pipeline for crawling, rendering and indexing a page.
+  </figcaption>
+</figure>
+
+The process works like this:
+
+1. Googlebot queues URLs for crawling.
+2. It then fetches the URLs with an HTTP request based on the [crawl budget](https://webmasters.googleblog.com/2017/01/what-crawl-budget-means-for-googlebot.html).
+3. Googlebot's next stage is scanning the HTML for links and queues the discovered links for crawling.
+4. Googlebot then queues the page for rendering.
+5. As soon as possible, a headless Chromium instance renders the page (JavaScript is executed here).
+6. Googlebot uses the rendered HTML to index the page.
+
+Your technical setup can influence the process of crawling, rendering, and indexing. For example, slow response times or server errors can impact the [crawl budget](https://webmasters.googleblog.com/2017/01/what-crawl-budget-means-for-googlebot.html) and links that are only rendered with JavaScript can lead to a slower discovery of these links.
+
+## Choose the right rendering strategy for your web app
+
+The default rendering strategy for single-page apps today is client-side rendering. The HTML loads the JavaScript which generates the content in the browser as it executes. 
+
+<figure class="w-figure">
+  <img src="spa-kittens.png" alt="A code box showing HTML that loads some scripts. A screenshot of a web page on mobile that shows placeholder images while loading the actual content.">
+  <figcaption class="w-figcaption">
+    The rendering strategy influences the performance and robustness of your web apps.
+  </figcaption>
+</figure> 
+
+The rendering strategy of your website influences performance. It also has an impact on how bots like Googlebot interact with it. Server-side rendering or pre-rendering can improve the performance for users and crawlers. Those strategies execute JavaScript on the server to generate the initial HTML content. The browser can start rendering the HTML as it arrives over the network, making the page load faster.  The [rendering session at I/O](https://www.youtube.com/watch?v=k-A2VfuUROg)  or [the blog post about rendering on the web](https://developers.google.com/web/updates/2019/02/rendering-on-the-web) shows how server-side rendering and hydration can improve the performance and user experience of web apps with more examples of how these strategies are implemented.
```

Comment:
```suggestion
The rendering strategy of your website influences performance. It also has an impact on how bots like Googlebot interact with it.

If you're free to choose your rendering strategy, consider server-side rendering or pre-rendering. They execute JavaScript on the server to generate the initial HTML content, which can improve performance for both users and crawlers. These strategies allow the browser to start rendering HTML as it arrives over the network, making the page load faster.  The [rendering session at I/O](https://www.youtube.com/watch?v=k-A2VfuUROg)  or [the blog post about rendering on the web](https://developers.google.com/web/updates/2019/02/rendering-on-the-web) shows how server-side rendering and hydration can improve the performance and user experience of web apps and provides more examples of how these strategies can be implemented.
```
Edits to help establish the distinction with the subsequent paragraph (can you pick your rendering strategy, or are you stuck with what you have?). Beyond that, need to define hydration. Or, even better, perhaps just say there are several strategies for increasing the amount of rendering done server-side, and that they can improve performance for both users and bots if used well. That way, you don't have to get into the weeds of the different server-side or hybrid strategies.